packages/frontend/src/fetcher/on-chain/index.ts

```diff
@@ -70,38 +66,26 @@ class Fetcher implements IPartialFetcher {
         if (reopenedQuestionId && reopenedQuestionId !== BYTES32_ZERO)
             finalQuestionId = reopenedQuestionId;
 
-        const {
-            content_hash,
-            arbitrator,
-            opening_ts,
-            timeout,
-            finalize_ts,
-            is_pending_arbitration,
-            bounty,
-            best_answer,
-            history_hash,
-            bond,
-            min_bond,
-        } = (await realityContract.read.questions([
+        const realityV3Question = await realityContract.read.questions([
             finalQuestionId,
-        ])) as unknown as OnChainRealityQuestion;
+        ]);
 
         return {
             id: finalQuestionId,
             reopenedId: questionId === finalQuestionId ? undefined : questionId,
-            historyHash: history_hash,
             templateId: Number(templateId),
             content: question,
-            contentHash: content_hash,
-            arbitrator,
-            timeout,
-            openingTimestamp: opening_ts,
-            finalizationTimestamp: finalize_ts,
-            pendingArbitration: is_pending_arbitration,
-            bounty,
-            bestAnswer: best_answer,
-            bond,
-            minBond: min_bond,
+            contentHash: realityV3Question[0],
```

Comment:
Doesn't `viem` infer the types here? Why do we have an array?

Reply:
Yes if infers the types, as an array; I don't know why it doesn't return an object though.

Reply:
https://github.com/wagmi-dev/viem/blob/main/.github/FAQ.md#why-is-a-contract-function-return-type-returning-an-array-instead-of-an-object

Reply:
> https://github.com/wagmi-dev/viem/blob/main/.github/FAQ.md#why-is-a-contract-function-return-type-returning-an-array-instead-of-an-object

Nice! Thank you @jxom

Reply:
Fixed in 5d8d93a513ceed84872c48bbf8bc27ac0fc558a7.